doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,203 @@
+These parameters were added prior to the creation of the ``Axes.bxp`` method,
+which draws boxplots from a list of dictionaries containing the relevant
+statistics. Matplotlib also provides a function that computes these
+statistics via ``cbook.boxplot_stats``. Note that advanced users can now
+either a) write their own function to compute the stats required by
+``Axes.bxp``, or b) modify the output returned by ``cbook.boxplots_stats``
+to fully customize the position of the artists of the plots. With this
+flexibility, the parameters to manually specify only the medians and their
+confidences intervals remain for backwards compatibility.
+
+Around the same time that the two roles of ``Axes.bxplot`` were split into
+``cbook.boxplot_stats`` for computation and ``Axes.bxp`` for drawing, both
+``Axes.boxplot`` and ``Axes.bxp`` were written to accept parameters that
+individually toggle the drawing of all components of the boxplots, and
+parameters that individually configure the style of those artists. However,
+to maintain backwards compatibility, the ``sym`` parameter (previously used
+to specify the symbol of the fliers) was retained. This parameter itself
+requires fairly complex logic to reconcile the ``sym`` parameters with the
+newer ``flierprops`` parameter at the default style specified by ``matplotlibrc``.
+
+This MEP seeks to dramatically simplify the creation of boxplots for
+novice and advanced users alike. Importantly, the changes proposed here
+will also be available to downstream packages like seaborn, as seaborn
+smartly allows users to pass arbitrary dictionaries of parameters through
+the seaborn API to the underlying matplotlib functions.
+
+This will be achieved in the following way:
+
+  1. ``cbook.boxplot_stats`` will be modified to allow pre- and post-
+     computation tranformation functions to be passed in (e.g., ``np.log``
+     and ``np.exp`` for lognormally distributed data)
+  2. ``Axes.boxplot`` will be modified to also accept and naïvely pass them
+     to ``cbook.boxplots_stats`` (Alt: pass the stat function and a dict
+     of its optional parameters).
+  3. Outdated parameters from ``Axes.boxplot`` will be deprecated and
+     later removed.
+
+Implementation
+==============
+
+Passing transform functions to ``cbook.boxplots_stats``
+-------------------------------------------------------
+
+This MEP proposes that two parameters (e.g., ``transform_in`` and
+``transform_out`` be added to the cookbook function that computes the
+statistics for the boxplot function. These will be optional keyword-only
+arguments and can easily be set to ``lambda x: x`` as a no-op when omitted
+by the user. The ``transform_in`` function will be applied to the data
+as the ``boxplot_stats`` function loops through each subset of the data
+passed to it. After the list of statistics dictionaries are computed the
+``transform_out`` function is applied to each value in the dictionaries.
```

Comment:
I think we're saying the same thing, but AFAIK, you can't pass a dictionary to e.g., `np.exp`. So you have to loop through the items. That's all I mean.

```python
for key, value in stat_dict.items():
    if key != 'label':
        stat_dict[key] = transform_out(value)
```